### ISSUE TITLE
Undefined export worker for app dir, this is a bug in Next.js.

### ISSUE THREAD

**Issue by mmrath** [vercel/next.js#49367]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:58 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6020
    Binaries:
      Node: 19.8.1
      npm: 9.5.1
      Yarn: 1.22.19
      pnpm: 8.1.0
    Relevant packages:
      next: 13.4.1
      eslint-config-next: 13.4.1
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue

NA

### To Reproduce

I am not too sure but definitely upgrading next from 13.3.1 to  both 13.3.2 or 13.4.1 causes this. Everything is good if I go back to 13.3.1

### Describe the Bug

Error: invariant: Undefined export worker for app dir, this is a bug in Next.js.
    at /repo/web/nextjs/node_modules/.pnpm/next@13.4.1_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/export/index.js:506:27
    at Span.traceAsyncFn (/repo/web/nextjs/node_modules/.pnpm/next@13.4.1_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/trace/trace.js:103:26)
    at /repo/web/nextjs/node_modules/.pnpm/next@13.4.1_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/export/index.js:502:35
    at Array.map (<anonymous>)
    at /repo/web/nextjs/node_modules/.pnpm/next@13.4.1_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/export/index.js:499:41
    at async Span.traceAsyncFn (/repo/web/nextjs/node_modules/.pnpm/next@13.4.1_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/trace/trace.js:103:20)
    at async /repo/web/nextjs/node_modules/.pnpm/next@13.4.1_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/index.js:1887:17
    at async Span.traceAsyncFn (/repo/web/nextjs/node_modules/.pnpm/next@13.4.1_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/trace/trace.js:103:20)
    at async build (/repo/web/nextjs/node_modules/.pnpm/next@13.4.1_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/index.js:143:29)

### Expected Behavior

build should finish

**Comment by mmrath** [comment]
May be a duplicate
https://github.com/vercel/next.js/issues/49059

**Comment by NiclasHaderer** [comment]
This was fixed for me in `^13.4.2-canary.4`

**Comment by timneutkens** [comment]
Going to close this based on @NiclasHaderer's comment as there is no reproduction to verify.
